### PR TITLE
 GEODE-8870: Removes GFE_5* and GFE_6*

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by DonalEvans — CHANGES_REQUESTED** [review]
Just a couple of little code clean-up suggestions and a typo.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommandQuery.java
Comment (echobravopapa) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommandQuery.java:290]: those must have been considered VERY important comments ... I mean nice deletion...

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommandQuery.java
Comment (echobravopapa) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommandQuery.java:177]: presume this means all existing/supported client versions can handle the key...

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifier.java
Comment (echobravopapa) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifier.java:486]: 💯

**Review by echobravopapa — COMMENTED** [review]
Appreciate the clean up, left a couple questions, probably @Bill should have a look

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java
Comment (Bill) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java:1699]: Ah yes you fixed a bug here!

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/SocketMessageWriter.java
Comment (Bill) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/SocketMessageWriter.java:50]: It appears to me (from looking at `ClientCacheNotifier`) that `clientVersion` is _probably_ never `null`. Are you confident that the null check was superfluous?

It's _almost_ not used anymore. But it is still used.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/Get70.java
Comment (Bill) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/Get70.java:113]: `regionName == null` is always `true` ✓

**Review by Bill — CHANGES_REQUESTED** [review]
Great PR. You took a big bite out!

I've got a few questions below…

File: geode-core/src/main/java/org/apache/geode/internal/cache/persistence/DiskInitFileParser.java
Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/persistence/DiskInitFileParser.java:476]: If it is redundant it is noise.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommand.java
Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommand.java:1309]: Yes, IJ wants concise readable code. I subscribe to the same.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommandQuery.java
Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommandQuery.java:177]: Yes, the corresponding method `sendCqResultsWithKey()` would now always return true. I refactored it out based on static analyzer.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientHealthMonitor.java
Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientHealthMonitor.java:667]: Yes, seems odd we track the number. I wonder if there is really any savings by this complexity. Outside my scope but at least now it is obvious what this counter is for.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/SocketMessageWriter.java
Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/SocketMessageWriter.java:50]: I am confident that if it was `null` we would have much bigger problems since all current client versions expect this content to be sent. 

We could change the logic for the check at the end against `GEODE_1_5_0` to account for `null`.

Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/SocketMessageWriter.java:50]: Even there though, the clients expecting this `int` will get pretty upset if they don't get it.

Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/SocketMessageWriter.java:50]: I think it is best that an exception occur here rather than unknown behavior from silently missing `int` in the message.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java
Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java:608]: It raises a good question. Returning or throwing form a finally block is strongly discouraged. Given the lack of comments it was hard to infer the intent. Given your concern that it may have been intentionally discarding exceptions there is enough reasonable doubt to undo this change. It needs more investigation to understand the intentions.

**Review by DonalEvans — APPROVED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/SocketMessageWriter.java
Comment (Bill) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/SocketMessageWriter.java:50]: I have manually verified that `clientVersion` comes from `ClientRegistrationMetadata` which acquires it in `getAndValidateClientVersion()` called from `initialize()` which is called before this method (`writeHandshakeMessage`). When there is a failure to get `clientVersion` this method is not called.

I see a [Geode dev list discussion from 2017](https://markmail.org/search/?q=list%3Aorg.apache.geode.dev+order%3Adate-backward+notnull#query:list%3Aorg.apache.geode.dev%20order%3Adate-backward%20notnull+page:1+mid:daov733nyw5xbhps+state:results) where use of the JetBrains `@NotNull` annotations were discussed with a couple folks agreeing they would be good for Geode. I approve this code as-is, but if you'd like to add some `@NotNull` annos here that would be worth an extra slice of cake 🍰 .

**Review by Bill — APPROVED** [review]
Approve!

**Pull request merged**
